design/ceph/mon-storage-class.md

```diff
@@ -0,0 +1,167 @@
+      # The default is "host", which means that each OSD must be on a different node and you would need at least two nodes per zone.
+      # If the subFailureDomain is set to "osd", the OSDs would be allowed anywhere in the same zone including on the same node.
+      # If set to "rack" or some other intermediate failure domain, those labels would also need to be set on the nodes where
+      # the osds are started.
+      subFailureDomain: host
+      zones:
+        - name: us-east-2a
+          arbiter: true
+        - name: us-east-2b
+        - name: us-east-2c
+    volumeClaimTemplate:
+      spec:
+        storageClassName: gp2
+        resources:
+          requests:
+            storage: 10Gi
+```
+
+The operator will track which mons belong to which zone. The zone assignment for each mon
+will be stored in the `rook-ceph-mon-endpoints` configmap in the same data structure where the host
+assignment is stored for mons with node affinity.
+
+For example, if the zones are called
+"arbiter", "zone1", and "zone2", the configmap would contain:
+
+```
+data:
+  data: a=10.99.109.200:6789,b=10.98.18.147:6789,c=10.96.86.248:6789,d=10.96.86.249:6789,e=10.96.86.250:6789
+  mapping: '{"node":{"a":"arbiter","b":"zone1","c":"zone1","d":"zone2","e":"zone2"}}'
+  maxMonId: "4"
+```
+
+If no zones are listed in the `spec.mon.zones`, It will default to arbitrary
+deployment without zonal awareness.
+
+### Mon Failover
+
+Mon failover will replace a mon in the same zone where the previous mon failed.
+
+### Mon Storage
+
+The mons can all be backed by a host path or a PVC, similar to any other Rook cluster.
+
+In this example, all the mons will be backed by the same PVC:
+
+```yaml
+  mon:
+    count: 5
+    allowMultiplePerNode: false
+    zones:
+    - name: arbiter
+      arbiter: true
+    - name: a
+    - name: b
+    volumeClaimTemplate:
+      spec:
+        storageClassName: gp2
+        resources:
+          requests:
+            storage: 10Gi
+```
+
+It is possible that the mons will need to be backed by a different type of storage in different zones.
+In that case, a volumeClaimTemplate specified under the zone will override the default storage setting for
+the mon.
+
+For example, the arbiter could specify a different backing store in this manner:
+
+```yaml
+  mon:
+    count: 5
+    allowMultiplePerNode: false
+    zones:
+    - name: arbiter
+      arbiter: true
+      volumeClaimTemplate:
+        spec:
+          storageClassName: alternative-storage
+            resources:
+              requests:
+                storage: 10Gi
+    - name: a
+    - name: b
+    volumeClaimTemplate:
+      spec:
+        storageClassName: gp2
+        resources:
+          requests:
+            storage: 10Gi
+```
+
+### Ceph Config
+
+The mons will be configured so Ceph will associate the mons with the correct failure domains.
+Rook will set each mon location to the zone name
+
+```console
+$ ceph mon set_location <mon> <zone>
```

Comment:
Does ceph use this info in some way? I thought this property was just for stretch clusters.